### PR TITLE
chore: Fix test coverage config

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Fix test coverage configuration. Not all files were collected for the coverage

Related links, issue #, if available: n/a

### How has this been tested?

Codecov report speaks for itself. 39.30% on the mainline and 83.73% in this PR

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,7 +75,7 @@
     "@typescript-eslint/eslint-plugin": "^5.48.0",
     "@typescript-eslint/parser": "^5.48.0",
     "@vitejs/plugin-react": "^4.2.1",
-    "@vitest/coverage-istanbul": "^1.1.1",
+    "@vitest/coverage-v8": "^1.3.1",
```
Comment (just-boris) [inline package.json:78]: Istanbul plugin had some issues instrumenting our code. v8 works better and also it is the default in vitest now

File: package.json [hunk 2]
```diff
@@ -22,10 +22,10 @@
     "preview": "vite preview",
     "start": "run-p start:server start:watch:ts start:watch:css",
     "start:server": "vite",
-    "start:watch:ts": "tsc --watch",
+    "start:watch:ts": "npm run build:src:js -- --watch",
     "start:watch:css": "chokidar \"./src/**/*.scss\" -c \"npm run build:src:css\"",
     "build:pkg": "node scripts/package-json.js",
-    "build:src:js": "tsc -p tsconfig.json",
+    "build:src:js": "tsc -p tsconfig.json --inlineSources --sourceMap",
```
Comment (just-boris) [inline package.json:28]: Source maps are required to resolve test coverage for imports from `lib/components/` path. Also, nice dev experience improvement for our consumers

File: package.json [hunk 3]
```diff
@@ -22,10 +22,10 @@
     "preview": "vite preview",
     "start": "run-p start:server start:watch:ts start:watch:css",
     "start:server": "vite",
-    "start:watch:ts": "tsc --watch",
+    "start:watch:ts": "npm run build:src:js -- --watch",
```
Comment (cansuaa) [inline package.json:25]: There is an additional `--` after build:src:js?

Comment (just-boris) [reply package.json:25]: Yes, this is the way it works: https://docs.npmjs.com/cli/v9/commands/npm-run-script

You split arguments to `npm` and arguments you want to pass down to the script, `tsc` in this case with double dash

Comment (cansuaa) [reply package.json:25]: Got it, thanks for explaining

**Review by cansuaa — APPROVED** [review]

**Pull request merged**